### PR TITLE
Fix onClick event for column button

### PR DESCRIPTION
**Pull request by dxvladislavvolkov**
### Cherry-picks
- #25644

Docs:
- https://github.com/DevExpress/devextreme-documentation/pull/5506

### CODE REVIEW

**Review by pomahtri — DISMISSED** [review]

File: packages/devextreme/js/ui/data_grid.d.ts [hunk 1]
```diff
@@ -2303,12 +2303,7 @@ export interface dxDataGridColumnButton<TRowData = any, TKey = any> extends Colu
     name?: DataGridPredefinedColumnButton | string;
     /**
      * @docid dxDataGridColumnButton.onClick
-     * @type_function_param1 e:object
-     * @type_function_param1_field component:dxDataGrid
-     * @type_function_param1_field model:object
-     * @type_function_param1_field event:event
-     * @type_function_param1_field row:dxDataGridRowObject
-     * @type_function_param1_field column:dxDataGridColumn
+     * @type_function_param1 e:{ui/data_grid:ColumnButtonClickEvent}
```
Comment (mpreyskurantov) [inline packages/devextreme/js/ui/data_grid.d.ts:2319]: Let's make a counterpart fix for TreeList and add doc comments for the ColumnButtonClickEvent type(s) to be published.
Blocked until the mono-repo CI fix.

**Review by mpreyskurantov — CHANGES_REQUESTED** [review]

File: packages/devextreme/js/ui/data_grid.d.ts (hunk 1 above)
Comment (mpreyskurantov) [reply packages/devextreme/js/ui/data_grid.d.ts:2319]: Moreover, TopicsGenerator fails without declaring the "ColumnButtonClickEvent" type linked via uid ref e:{ui/data_grid:ColumnButtonClickEvent}

Comment (mpreyskurantov) [reply packages/devextreme/js/ui/data_grid.d.ts:2319]: Discard TreeList. Add declarations for DataGrid.

**Review by mpreyskurantov — APPROVED** [review]

**Review by pomahtri — DISMISSED** [review]

**Review by OlegKipchatov — APPROVED** [review]

File: packages/devextreme-angular/src/ui/nested/base/text-editor-button-dxi.ts [hunk 2]
```diff
@@ -60,10 +61,10 @@ export abstract class DxiTextEditorButton extends CollectionNestedOption {
...
     }
 
-    get onClick(): Function {
+    get onClick(): ((e: ColumnButtonClickEvent) => void) {
         return this._getOption('onClick');
     }
-    set onClick(value: Function) {
+    set onClick(value: ((e: ColumnButtonClickEvent) => void)) {
         this._setOption('onClick', value);
     }
```
Comment (mpreyskurantov) [inline packages/devextreme-angular/src/ui/nested/base/text-editor-button-dxi.ts:69]: Confirm if this change ever has effect.

**Review by mpreyskurantov — CHANGES_REQUESTED** [review]

File: packages/devextreme-angular/src/ui/nested/base/text-editor-button-dxi.ts (hunk 2 above)
Comment (mpreyskurantov) [reply packages/devextreme-angular/src/ui/nested/base/text-editor-button-dxi.ts:69]: TypeScript checking does not work for nested options (at least without the @Input() decorator).

**Review by mpreyskurantov — APPROVED** [review]

**Pull request merged**